compiler/src/dotty/tools/dotc/CompilationUnit.scala

```diff
@@ -20,3 +23,19 @@ class CompilationUnit(val source: SourceFile) {
   /** Pickled TASTY binaries, indexed by class. */
   var pickled: Map[ClassSymbol, Array[Byte]] = Map()
 }
+
+object CompilationUnit {
+
+  /** Make a compilation unit for top class `clsd` with the contends of the `unpickled` */
+  def mkCompilationUnit(clsd: ClassDenotation, unpickled: Tree)(implicit ctx: Context): CompilationUnit = {
+    val unit1 = new CompilationUnit(new SourceFile(clsd.symbol.sourceFile, Seq()))
+    unit1.tpdTree = unpickled
+    force.traverse(unit1.tpdTree)
```

Comment:
Do you actually need to force the tree for the linker?

Reply:
When we first tried it with @DarkDimius we needed it. But it looks like we do not need it anymore.